promising/promise.py

```
@@ -425,31 +336,78 @@ def _activate(self) -> None:
-            #  when the whole python process exits ? We should somehow show the
-            #  errors to the user as soon as they happen (for all children, not
-            #  just the ones that make the parent wait).
-            await asyncio.gather(*promises_to_await, return_exceptions=True)
+        # TODO Make it possible to call this method from another thread
+        # TODO Ideally, a warning should be issued if any of the children are
+        #  configured with start_soon=False, because that would make it quite
+        #  easy to introduce deadlocks.
+        await asyncio.gather(*self.get_pending_children(), return_exceptions=raise_exceptions)
+
+    def _setup_start_soon(self, *, start_soon: bool | Sentinel, children_start_soon: bool | Sentinel) -> None:
+        from promising import should_start_soon_by_default  # noqa: PLC0415 (import-outside-top-level)
```

Ruff (0.15.0)

[warning] Unused `noqa` directive (non-enabled: `PLC0415`)

Remove unused `noqa` directive

(RUF100)